src/main/java/com/dev/cinema/controller/ShoppingCartController.java

```diff
@@ -34,15 +36,21 @@ public ShoppingCartController(ShoppingCartService shoppingCartService,
     }
 
     @PostMapping
-    public void addSession(@RequestParam Long userId, @RequestParam Long movieSessionId) {
-        User user = userService.getById(userId);
+    public void addSession(Authentication authentication, @RequestParam Long movieSessionId) {
+        UserDetails details = (UserDetails) authentication.getPrincipal();
+        String email = details.getUsername();
+        User user = userService.findByEmail(email).orElseThrow(()
+                -> new RuntimeException("Incorrect email or password"));
```

Comment:
let's just call `get()` method. NoSuchElementException could be handled in GlobalExceptionHandler (this is not a part of HW)

Reply:
@AntonPanchishko And here?